base/strings/util.jl

```diff
@@ -602,7 +602,7 @@ function ascii(s::String)
     end
     return s
 end
-@noinline __throw_invalid_ascii(s, i) = throw(ArgumentError("invalid ASCII at index $i in $(repr(s))"))
+@noinline __throw_invalid_ascii(s::String, i::Int) = throw(ArgumentError("invalid ASCII at index $i in $(repr(s))"))
```

Comment:
This seems like a mostly-unrelated change; is this just cleanup?  Do the type arguments help codegen because of `@noinline`?

Reply:
Yes, it's unrelated. Just good style to write the types since we know them exactly.